client/verta/verta/_internal_utils/model_dependencies.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+
+import ast
+import inspect
+import re
+from importlib.metadata import packages_distributions
+from typing import Callable, List, Set, Tuple, Type, get_type_hints
+from types import ModuleType
+from verta.registry import VertaModelBase
+
+
+def list_class_functions(model_class: Type[VertaModelBase]) -> List[Tuple[str, Callable]]:
+    """List all the functions present in the provided class object."""
+    return inspect.getmembers(model_class, predicate=inspect.isfunction)
+
+
+def list_modules_in_function_definition(func: Callable) -> List[ModuleType]:
+    """List all modules called within the body of the provided function."""
+    try:
+        _func = inspect.unwrap(func)  # returns last object in chain
+    except ValueError:  # if a cycle is encountered
```

Comment:
Just out of curiosity, did you encounter any particular cases with cycles while experimenting? I'm curious what libraries might have them‚ or if that even works in Python at all.

Reply:
I did not.  I'm basically just recycling [documentation](https://docs.python.org/3/library/inspect.html#inspect.unwrap) here.